src/Access/TaxonomyDeleteAccessCheck.php

```diff
@@ -0,0 +1,212 @@
+      $assigned_content = $this->isAssignedToContent($term);
+
+      /*
+       * If this term does not have users assigned to it for access
+       * control, and the term is not assigned to any pieces of content,
+       * it is OK to delete it.
+       */
+      if ($hasAccessControlMembers || $assigned_content) {
+
+        /** @var \Drupal\Core\Session\AccountProxyInterface $user */
+        $user = \Drupal::currentUser();
+
+        $override_allowed = $user->hasPermission('allow taxonomy term delete');
+
+        if ($assigned_content && !$override_allowed) {
+          drupal_set_message(t("The term %term is being used to tag content and may not be deleted.",
+            ['%term' => $term->getName()]), 'warning');
+          $retval = FALSE;
+        }
+        elseif ($assigned_content) {
+          drupal_set_message(t("The term %term is being used to tag content.",
+            ['%term' => $term->getName()]), 'warning');
+          $retval = TRUE;
+        }
+
+        if ($hasAccessControlMembers) {
+          drupal_set_message(t("The term %term is being used for access control and may not be deleted.",
+           ['%term' => $term->getName()]), 'warning');
+          $retval = FALSE;
+        }
+
+      }
+    }
+
+    return $retval;
+
+  }
+
+  /**
+   * Determines if this term has active members in it.
+   *
+   * @return bool
+   *   TRUE if the term has members, FALSE otherwise.
+   */
+  private function doesTermHaveMembers(TermInterface $term) {
+
+    /** @var array $sections */
+    $sections = $this->getActiveSections($term);
+
+    if (count($sections) > 0) {
+      return TRUE;
+    }
+
+    return FALSE;
+  }
+
+  /**
+   * Helper function to extract the entity for the supplied route.
+   *
+   * @return null|TermInterface
+   *   A term interface object if the term exists. NULL otherwise.
+   */
+  private function getRouteEntity() {
+    $route_match = \Drupal::routeMatch();
+    // Entity will be found in the route parameters.
+    if (($route = $route_match->getRouteObject()) && ($parameters = $route->getOption('parameters'))) {
+      // Determine if the current route represents an entity.
+      foreach ($parameters as $name => $options) {
+        if (isset($options['type']) && strpos($options['type'], 'entity:') === 0) {
+          $entity = $route_match->getParameter($name);
+          if ($entity instanceof TermInterface) {
+            return $entity;
+          }
+
+          // Since entity was found, no need to iterate further.
+          return NULL;
+        }
+      }
+    }
+  }
+
+  /**
+   * Inspect the given taxonomy term.
+   *
+   * This will determine if there are any active users assigned to it.
+   *
+   * @param \Drupal\taxonomy\TermInterface $term
+   *   The Taxonomy Term to inspect.
+   *
+   * @return array
+   *   An array of the users assigned to this section.
+   */
+  private function getActiveSections(TermInterface $term) {
+    $container = \Drupal::getContainer();
```

Comment:
the `workbench_access.user_section_storage` service (not the container) should be injected

Reply:
done.